Gmdx 117

bjdupuis opened the pull request:
Reopening this PR for publishing.

bjdupuis committed:
Getting NPM publishing going

bjdupuis committed:
Merge branch 'main' into GMDX-117

bjdupuis committed:
Use parameter for email list

Per Chris Rumpf's suggestion

bjdupuis committed:
Merge branch 'main' into GMDX-117

bjdupuis committed:
Parameter for NPM credentials and putting dry-run back to test

bjdupuis committed:
Putting in the gate for the release branch on publish step

bjdupuis committed:
Merge branch 'main' into GMDX-117

bjdupuis committed:
Remove the conditional on the branch.

bjdupuis committed:
Merge branch 'main' into GMDX-117

bjdupuis committed:
Re-open the PR

tehilaR2 commented on Jenkinsfile:
```diff
@@ -0,0 +1,135 @@
+  }
+
+  tools {
+    nodejs 'NodeJS 12.13.0'
```
i got an error on this one indicating nodejs is not a tool
is it working 4 you?

bjdupuis replied:
In the Jenkins environment the dry runs have been working well. I don't run a local Jenkins setup for testing given it really only _has_ to work in our Jenkins environment.

tehilaR2 commented on Jenkinsfile:
```diff
@@ -0,0 +1,135 @@
+
+    stage('Prep') {
+      steps {
+        sh "git clone --single-branch -b master --depth=1 git@bitbucket.org:inindca/npm-utils.git ${env.NPM_UTIL_PATH}"
```
deleteDir() should be added before this action in case the script gets replayed

bjdupuis replied:
I don't believe there's a case in the Jenkins environment where we'd be replaying the script within the workspace for a build invocation. Different attempts are rebuilds which have their own workspaces.

tehilaR2 commented on Jenkinsfile:
```diff
@@ -0,0 +1,135 @@
+                echo "npm-utils" >> .npmignore
+                npm publish 1>&2
+            '''
+            sshagent (credentials: ['3aa16916-868b-4290-a9ee-b1a05343667e']) {
```
this credentials are better be set as parameter

bjdupuis committed:
Testing new parameter for Github credentials

Adding back `dry-run` and removing the release branch check for testing

bjdupuis committed:
Ignore the Jenkinsfile for the NPM package.

bjdupuis committed:
Try again.

bjdupuis committed:
Again

Looking at the workspace on Jenkins the npm-utils isn't in the directory being published. Don't need to explicitly ignore it.

bjdupuis committed:
Looks ready

Remove testing
